### PR TITLE
Allow deployment for existing name prefix cluster if the instances are Terminated

### PR DESCRIPTION
**Pull request by ebenahar**
Addresses issue https://github.com/red-hat-storage/ocs-ci/issues/798

Signed-off-by: Elad Ben Aharon <ebenahar@redhat.com>

### CODE REVIEW

File: ocs_ci/deployment/aws.py [hunk 1]
```diff
@@ -89,7 +89,13 @@ def check_cluster_existence(self, cluster_name_prefix):
...
 
         """
-        if self.aws.get_instances_by_name_pattern(cluster_name_prefix):
+        instances = self.aws.get_instances_by_name_pattern(cluster_name_prefix)
+        instance_objs = [self.aws.get_ec2_instance(ins.get('id')) for ins in instances]
+        non_terminated_instances = [
+            ins for ins in instance_objs if ins.state
+            .get('Code') != constants.INSTANCE_TERMINATED
+        ]
+        if non_terminated_instances:
```
Comment (mbukatov) [inline ocs_ci/deployment/aws.py]: I would rather used an explicit check like `len(non_terminated_instances) > 0` ...

Comment (mbukatov) [reply ocs_ci/deployment/aws.py]: You could also consider logging the non terminated instances.

**Comment by ebenahar** [comment]
Deployment finished successfully here:
https://ocs4-jenkins.rhev-ci-vms.eng.rdu2.redhat.com/job/qe-deploy-ocs-cluster/1065/console

We're using a different config in Jenkins, that allows multiple clusters with the same name prefix.

As for local runs:

[deployment_logs.zip](https://github.com/red-hat-storage/ocs-ci/files/3642926/deployment_logs.zip)

This was executed while having a cluster with running instances, therefore, this PR doesn't break anything.

File: ocs_ci/deployment/aws.py (hunk 1 above)
Comment (ebenahar) [reply ocs_ci/deployment/aws.py]: Sorry @mbukatov, just now see your comments.
Regarding the first one, how is it different? if non_terminated_instances is basically the same as if the len of it is 0.
Regarding the second one, I'll raise a PR for improving the logging

Comment (ebenahar) [reply ocs_ci/deployment/aws.py]: https://github.com/red-hat-storage/ocs-ci/pull/804

**Pull request merged**